src/sql/src/plan/statement/ddl.rs

```diff
@@ -760,6 +770,79 @@ pub fn plan_create_source(
         None => scx.catalog.config().timestamp_interval,
     };
 
+    let source_desc = SourceDesc {
+        connection: external_connection,
+        encoding,
+        envelope: envelope.clone(),
+        metadata_columns: metadata_column_types,
+        timestamp_interval,
+    };
+
+    // Here we will store the available subsources for this ingestion. Sources like postgres and
+    // the load generator make additional streams available identified by their index and
+    // associated to a FullObjectName
+    let mut available_subsources = HashMap::new();
+    for (name, output_idx) in source_desc.subsource_outputs() {
+        let name = match normalize::full_name(name) {
+            Ok(name) => name,
+            Err(_) => sql_bail!("[internal error] sources must declare valid subsource names"),
```

Comment:
We can't use `soft_assert` here (or in the other cases) because soft assert doesn't unconditionally assert, or panic, which means that the code path having the `soft_assert` must return a correctly typed value and have the rest of the program continue.

In this case we can't really continue planning so the two options are returning from the planner with an error, or unconditionally panic. I think we don't want to unconditionally panic in this case

Reply:
Ah, sorry, I meant in addition to erroring, i.e. panic if it's reasonable to. Sounds like you'd rather not, so no need.